services/models/pod.go

```diff
@@ -93,3 +93,17 @@ func lookupImage(containerName string, containers []v1.Container) string {
 	}
 	return ""
 }
+
+func (pod *Pod) HasIstioSideCar() bool {
+	if len(pod.IstioContainers) > 0 {
+		return true
+	}
+	return false
+}
+
+func (pods Pods) HasIstioSideCar() bool {
+	for _, pod := range pods {
+		return pod.HasIstioSideCar()
```

Comment:
I think it is not working beacuse this returns only the first pod hasIstioSidecar value.

Reply:
I thought about.

```go
func hasIstioSideCar(pods []v1.Pod) bool {
	mPods := models.Pods{}
	mPods.Parse(pods)
	for _, pod := range mPods {
		if len(pod.IstioContainers) > 0 {
			return true
		}
	}
	return false
}
```

Reply:
It seems that we never check all pods... :( Changed it

Reply:
here it was the start of discussion @jotak ^^